skrf/circuit.py

```diff
@@ -140,10 +141,16 @@ def _get_nx():
         except ImportError as err:
             raise ImportError('networkx package as not been installed and is required.') from err
 
+    class _REDUCE_OPTIONS(TypedDict):
+        check_duplication: NotRequired[bool]
+        split_ground: NotRequired[bool]
+        max_nports: NotRequired[int]
+
     def __init__(self,
                  connections: list[list[tuple[Network, int]]],
                  name: str | None = None,
-                 auto_reduce: bool = False) -> None:
+                 *,
+                 auto_reduce: bool = False, **kwargs: Unpack[_REDUCE_OPTIONS]) -> None:
```

Comment:
why not simply

```suggestion
                 auto_reduce: bool = False, 
                 check_duplications: bool = True, 
                 split_grounds: bool = False, 
                 max_nports: int = 20) -> None:
```

without any `kwargs`?

Reply:
1. The parameters in `kwargs` are not essential for constructing a `Circuit` instance; they are only used when `auto_reduce=True`. Including them directly as **optional arguments** may be misleading, as it could suggest that these parameters are required for creating the `Circuit` instance, similar to other mandatory parameters, such as `name` and `connections`.

2. However, I realize that if a user provides these parameters via `kwargs` but forgets to set `auto_reduce=True`, their intention is likely to use the `reduce_circuit` method. Therefore, it would be reasonable to automatically enable `auto_reduce` based on the presence of relevant parameters in `kwargs`, ensuring that `reduce_circuit` is activated appropriately. 

For this reason, I believe placing these parameters in `kwargs` rather than listing them directly as **optional arguments** better supports the second point and avoids additional confusion. 

I’d love to hear your thoughts on this matter. Your feedback and suggestions would be greatly appreciated.

Reply:
Ok, I'm fine :) 
let's merge this.